### ISSUE TITLE
Having the same REPL on 20x4 LCD screen?

### ISSUE THREAD

**Issue by vitasam** [lerouxb/replpaper#1]
Hi,
great project! 
I'm playing with one of those extremely cheap I2C 20x4 LCD display, connected to esp32. Just out of curiosity - how easy (or difficult) will be to modify your code to support native 20x4 text mode?

**Comment by lerouxb** [comment]
The textbuffer.py file should work, but the rest is pretty tied to the e-ink screen I'm using and the exact keyboard I built.

I've shifted my focus to https://github.com/lerouxb/dumbdumb which is a more generic serial vt100 subset terminal so it will work with any/all REPLs that speak serial. ie. micropython, basic, various javascripts, lisps, forths, etc. Much more efficient c imlementation of textbuffer there too.

**Comment by vitasam** [comment]
Hi,
I got your point. I did quick experiments with esp32 and FBConsole (micropython) on ILI9341 TFT 320x240 display - performance is not for real applications, just for fun. "Pure" VT100 implementation will be much better. 

Some time ago I did stm32BASIC device with 20x4 LCD and PS/2 keyboard on the cheapest "bluepill" stm32 board:
https://youtu.be/Akqj6i8sFY8

It was based on libopencm3 STM32 opensource library. The project goes nowhere due to zero interest from communities.
